### ISSUE TITLE
3.11.1 discussions

### ISSUE THREAD

**Issue by joshpar** [supercollider/supercollider#4790]
## Discussion of 3.11.1 release issues - Pre-alpha

Developer discussion

Pre-planning. Pre-Alpha Stage as of 2020-03-01

See project board for status of PRs for this release.

**Comment by salkin-mada** [comment]
I would love if fixing native Wayland support #4544 could be considered for 3.11.1

**Comment by joshpar** [comment]
Discussions are started. Targeting a mid-April 2020 RC, end of April release.

**Comment by brianlheim** [comment]
based on the discussion in #4544, i would suggest that instead of trying to solve that issue (which is according to current knowledge, impossible), we at least document it as a known issue and have the plugins report it when they are loaded.

**Comment by brianlheim** [comment]
#4136 might be a stretch goal for 3.11.1, we should solve it soon i think

**Comment by dyfer** [comment]
We could consider cherry-picking #4850. It would make SC GUI useable on 4k screens (200% scaling).

(fractional scaling like 125% will be supported in 3.12).

**Comment by brianlheim** [comment]
seems fine to me, i added it to the cherry-pick list. what about #4846?

**Comment by elgiano** [comment]
Yes please, the two make each other better :)

**Comment by dyfer** [comment]
@elgiano do you think it's good to have #4844 without qt 5.14?

**Comment by elgiano** [comment]
Yes: #4846 removes the rounding from the only place where devicePixelRatio is rounded. It aligns QcCanvas with Widgets in the way they manage fractional scaling. The requirement for Qt5.14 in #4850 is to access different policies for rounding pixels, not the scaling factor.
That said, I'm never 100% sure, especially when it comes to Windows, do you see any potential issue there?
I'm trying to build again with Qt5.9 on Linux just to double-check

**Comment by dyfer** [comment]
@elgiano thanks for the explanation - this makes sense. If you can check the 5.9 build that would be great, but it does seem that adding #4846 to cherry pick is desirable.

**Comment by brianlheim** [comment]
i am going to cherry-pick #4846. if any issues come up in the rc we can revert it.

**Comment by brianlheim** [comment]
in the meeting yesterday we decided that after the cherry-picks (#4865 ) and scel submodule update (#4855 ) we can do the 3.11.1 release candidate. this means that alert sound with window in focus #4826 and pure-wayland support #4544 are postponed to a later release. i've updated the project board accordingly.

**Comment by brianlheim** [comment]
would really recommend including https://github.com/supercollider/supercollider/pull/4875 in the 3.11.1 release - it's a very simple fix and without it non-GUI builds are mildly broken.

**Comment by joshpar** [comment]
I say go for it.

**Comment by dyfer** [comment]
> I say go for it.

done!

**Comment by dyfer** [comment]
It seems that `String -runInTerminal` is broken in 3.11. Would we want to fix that in 3.11.1? https://github.com/supercollider/supercollider/pull/4880

**Comment by brianlheim** [comment]
i'm fine with it going into 3.11.1 but it's not a must-have in my opinion. i just reviewed it

**Comment by jamshark70** [comment]
If runInTerminal worked in 3.10.x and broke in 3.11.0 (which seems to be what the issue report says), I'd vote for calling it a regression issue.

**Comment by brianlheim** [comment]
yes it's obviously a regression, just not essential to fix in the patch release that is imminent

**Comment by brianlheim** [comment]
#4880 and #4865 are now both merged. #4855 is the last required PR for 3.11.1!

**Comment by dyfer** [comment]
I merged #4855

**Comment by dyfer** [comment]
I just wanted to put https://github.com/supercollider/supercollider/pull/4927 on people's radar - do we want to cherry-pick it?

**Comment by telephon** [comment]
Should we add #4659 to the list, please? It is an annoying and confusing error with a lot of impact.

**Comment by patrickdupuis** [comment]
Can these go into `3.11.1`?
- (cherry-pick into 3.11) https://github.com/supercollider/supercollider/pull/4938
- (rebase on 3.11) https://github.com/supercollider/supercollider/pull/4939

**Comment by eleses** [comment]
> I just wanted to put #4927 on people's radar - do we want to cherry-pick it?

Segfaulting on non-IP-configured interfaces (which that seems to be about) can't be a good thing.

**Comment by patrickdupuis** [comment]
I've just opened a bunch of cherry-pick PRs for `3.11.1`. Some of them I'm not sure about. Please close them if they shouldn't be included in the release.

**Comment by brianlheim** [comment]
thanks @patrickdupuis !

**Comment by brianlheim** [comment]
@patrickdupuis i just merged all the CI-passing and non functional (i.e. documentation changes) cherry-picks. the remainder i approved and i'm just waiting on CI to finish before merging. thanks again!

**Comment by brianlheim** [comment]
all those cherry-picked PRs are now merged, although there were two PRs that weren't cherry-picked in https://github.com/supercollider/supercollider/projects/19, not sure if you missed them or didn't have time for them @patrickdupuis . do you want to do that or should i?

**Comment by brianlheim** [comment]
as for some of my open PRs:

- https://github.com/supercollider/supercollider/pull/5046 : would like to wait until 3.11.2 for this one unless we are going from 3.11.1 to 3.12
- https://github.com/supercollider/supercollider/pull/5045 : a harmless documentation PR that would be nice to include in 3.11.1

**Comment by patrickdupuis** [comment]
@brianlheim #5045 is good to merge

#4939 needs to be reviewed again since I made some changes. @joshpar I think it should go into `3.11.1`.

**Comment by patrickdupuis** [comment]
#4990 is aimed at the `3.11` branch. Seems important for Linux packaging in Fedora at least. What needs to be done in order to have it merged?

**Comment by brianlheim** [comment]
#4990, #5045, #5046, #4939 have all been merged. i am going to make a PR momentarily for cherry-picking the remaining PRs on the project board into 3.11. after that, i don't want to merge anything more into 3.11 if we can help it. i will make an announcement when we are formally beginning the release process.

**Comment by brianlheim** [comment]
#5115 is the cherry-pick PR. i am also creating small PRs as I go through the release process steps, would appreciate if people can give those high priority! ^^

**Comment by brianlheim** [comment]
#5115 was merged. that's everything from the project board and this thread.

as i said in my announcement yesterday, i am planning to make the release next weekend. as far as i can tell there have been no show-stopping issues since we last agreed to start a release. so, i'd like to formally start the release process; i.e., no new PRs aside from those necessary for the release and those that fix regressions from 3.11.0 to 3.11.1-rc1 will be merged into 3.11 branch until after 3.11.1 is released. if anyone has a concern about that please let me know.

**Comment by jamshark70** [comment]
Could #5113 go in? Simple fix of a clearly wrong behavior, and I'd opened the PR a couple days before closing the release. (OTOH nobody including myself noticed the bug for years, so it isn't exactly life or death and I won't fuss if it has to wait. But it'll take one minute to review, so why not ask?) Thanks.

**Comment by brianlheim** [comment]
whoops, accidentally hit close.

that PR doesn't seem to fix a show stopping bug, so if it gets merged before i start the release process, i don't mind adding it to the changelog. otherwise it can go in 3.11.2.

**Comment by brianlheim** [comment]
@jamshark70 also want to add that i'd really like to be able to review more of the patterns lib PRs. right now they're outside of my area of expertise and i don't think i can spend the time to understand them and review them confidently while also doing the release + paying job stuff, but with you and Julian having less time to spend on SC these days i'd like to be able to help out more with those.

**Comment by brianlheim** [comment]
hey all,

we are now entering release mode for 3.11.1, and so the 3.11 branch is closed for everything except:
- critical bugfixes
- fixes for regressions discovered during the release candidate period

i expect to have 3.11.1-rc1 released this weekend, stay tuned for more announcements.

**Comment by brianlheim** [comment]
3.11.1-rc1 is out! we'll wait 2 weeks for rc testing, so the 3.11.1 release will happen on or after **August 20, 2020**.

**Comment by brianlheim** [comment]
@dvzrv can you check the PGP signature on this RC? i used my personal key and just want to make sure this is what works for you. don't want to cause another scare signing scare :)

**Comment by brianlheim** [comment]
since there have been no reports of major issues or regressions in the last 2 weeks, i'm planning to make the 3.11.1 release soon! will get started on the process tonight.

**Comment by brianlheim** [comment]
they have a fix but it will require modifying the CI config until the change is merged, which seems like it could take awhile (https://github.com/travis-ci/dpl/pull/1208), last PR merged to that repo took 6 weeks.

i am going to:
1. trigger the original build one last time to see if it will work
2. failing that, trigger a "custom build" of `master`, which will allow me to enter a custom yml config. i will have that build deploy to S3, then download the zip file and continue with the release that way

**Comment by brianlheim** [comment]
alright, that seemed to work! proceeding with the release.

**Comment by brianlheim** [comment]
#5147 is the last step in the release!

**Comment by brianlheim** [comment]
these PRs also need to be merged for the release:
- https://github.com/supercollider/supercollider.github.io/pull/221
- https://github.com/supercollider/sc3-plugins/pull/297
